src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -420,23 +484,29 @@ def register_options(cls, register):
              help='Use these javac plugins.')
     register('--javac-plugin-args', advanced=True, type=dict, default={}, fingerprint=True,
              help='Map from javac plugin name to list of arguments for that plugin.')
+    cls.register_jvm_tool(register, 'javac-plugin-dep', classpath=[],
+                          help='Search for javac plugins here, as well as in any '
+                               'explicit dependencies.')
 
     register('--scalac-plugins', advanced=True, type=list, fingerprint=True,
              help='Use these scalac plugins.')
     register('--scalac-plugin-args', advanced=True, type=dict, default={}, fingerprint=True,
              help='Map from scalac plugin name to list of arguments for that plugin.')
-
-    # Scalac plugin jars must already be available at compile time, because they need to be listed
-    # on the scalac command line. We search for available plugins on the tool classpath provided
-    # by //:scalac-plugin-jars.  Therefore any in-repo plugins must be published, so they can be
-    # pulled in as a tool.
-    # TODO: Ability to use built in-repo plugins via their context jars.
-    cls.register_jvm_tool(register, 'scalac-plugin-jars', classpath=[])
+    cls.register_jvm_tool(register, 'scalac-plugin-jars', classpath=[],
+                          removal_version='1.5.0.dev0',
+                          removal_hint='Use --compile-zinc-scalac-plugin-dep instead.')
```

Comment:
Hmmm, I wondered about this, and I'm still not sure.

Technically, plugins aren't part of the Scala platform - they're not part of any spec. Rather they are a detail of scalac.  If we one day switched to using a different compiler, it might not have plugins at all. So to me that makes it look more like a feature of our compilation task than of ScalaPlatform, which is mostly for specifying versions of the scala runtime.

It's true that we mention the scala compiler version in ScalaPlatform, and that scalac is closely tied to the runtime, but I'd rather not add to that confusion.

Similar considerations might  apply to Java, although the (poorly-named, my bad) Java subsystem doesn't specify anything about runtimes, just javac. So really it should be called Javac, and then I suppose it might be a sensible place for plugin settings.

Nonetheless, it really seems like more of a configuration of our compilation task, but I could be convinced otherwise.  I'm fuzzy on this one.

Thoughts? Other reviewers want to chime in?